atom/browser/ui/win/thumbar_host.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2015 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/ui/win/thumbar_host.h"
+
+#include <shobjidl.h>
+
+#include "base/win/scoped_comptr.h"
+#include "base/win/win_util.h"
+#include "base/win/wrapped_window_proc.h"
+#include "base/strings/utf_string_conversions.h"
+#include "third_party/skia/include/core/SkBitmap.h"
+#include "ui/gfx/icon_util.h"
+#include "ui/gfx/win/hwnd_util.h"
+
+namespace atom {
+
+namespace {
+
+// The thumbnail toolbar has a maximum of seven buttons due to the limited room.
+const int kMaxButtonsCount = 7;
```

Comment:
Yeah, it's illustrated in https://msdn.microsoft.com/en-us/library/windows/desktop/dd378460(v=vs.85).aspx#thumbbars. And I have made more comment here.